### PR TITLE
fix(hello-world): add CI pipeline for Linux and Windows

### PR DESCRIPTION
**Pull request by 0xE282B0**
Added the PowerShell scripts suggested by @axtens and created a GitHub actions pipeline to test them with Windows.
Unfortunately the cobol compiler is still missing some env vars.

### CODE REVIEW

**Comment by iHiD** [comment]
Just as a sanity check here - the eventual test runner will need to run on Linux, so is it worth going down that route in these files too, rather than Powershell?

(cc @erikschierboom)

**Comment by axtens** [comment]
I'm expecting that we will always support CLI even if GUI ends up being the usual method. Unlike Guido van Rossum, I'm not interested in being Benevolent Dictator for Life, so if alienating Windows CLI users is the go ...

**Comment by KTSnowy** [comment]
Wait, does the test runner always run on Exercism's Linux servers or does it run locally on the student's machine as well?

**Comment by axtens** [comment]
Two different modes of operation, @KTSnowy . GUI and CLI.

**Comment by 0xE282B0** [comment]
@iHiD, The shell scripts for Mac and Linux are already running. That's what I'm mainly working with. But I think that Windows CLI users are a group that's worth targeting. Especially if the effort is only to place a `test.ps1` file in the exercise folder.

**Comment by ErikSchierboom** [comment]
> Just as a sanity check here - the eventual test runner will need to run on Linux, so is it worth going down that route in these files too, rather than Powershell?

Ever since PowerShell started using .NET Core (now rebranded .NET), PowerShell has been able to run on any platform that can run .NET (which are most platforms). So having a PowerShell script is fine IMHO.

**Review by ErikSchierboom — APPROVED** [review]
LGTM

Side note: it is possible to combine the two workflows into a single workflow using some conditional logic. This makes it look a bit nicer in the github actions UI. See https://github.com/exercism/configlet/blob/main/.github/workflows/tests.yml#L44 for an example.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,27 +13,23 @@
...
 
-name: <track> / Test
+name: COBOL / Test (Linux)
 
-on:
-  push:
-    branches: [main]
-  pull_request:
-  workflow_dispatch:
+on: [push]
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:18]: Why did you change the original settings?

**Comment by ErikSchierboom** [comment]
> Wait, does the test runner always run on Exercism's Linux servers or does it run locally on the student's machine as well?

The test runner _always_ runs on Exercism's linux servers.

File: .github/workflows/test.yml (hunk 1 above)
Comment (0xE282B0) [reply .github/workflows/test.yml:18]: Just for test purposes.

File: exercises/practice/hello-world/test.ps1 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+if (![System.IO.File]::Exists("$PSScriptRoot/bin/cobolcheck.exe")){
+  Write-Output "Cobolcheck not found, try to fetch it."
+  & "$PSScriptRoot/bin/fetch-cobolcheck.ps1"
+}
+
+Write-Output "Run cobolcheck."
+cd $PSScriptRoot
+& bin/cobolcheck.exe -p hello-world
```
[the remark this answers is not in the document]
Comment (0xE282B0) [reply exercises/practice/hello-world/test.ps1:8]: Ok, the main problem is that the testing framework always returns with zero RC. I follow up on that here: https://github.com/openmainframeproject/cobol-check/issues/224

File: .github/workflows/test.yml (hunk 1 above)
Comment (0xE282B0) [reply .github/workflows/test.yml:18]: I need to move the CI pipeline out of this PR since it is getting unnecessarily complex otherwise. 
But I'll keep that in mind!

File: exercises/practice/hello-world/src/hello-world.CBL [hunk 3]
```diff
@@ -5,7 +5,6 @@
         WORKING-STORAGE SECTION.
         01 WS-RESULT PIC X(13).
         PROCEDURE DIVISION.
-            GOBACK.
```
Comment (0xE282B0) [inline exercises/practice/hello-world/src/hello-world.CBL:8]: @KTSnowy, according to https://github.com/openmainframeproject/cobol-check/issues/223 the `GOBACK` is not needed and the test is running fine without it. So, I removed it.

**Comment by 0xE282B0** [comment]
@axtens, can you try the PowerShell scripts?
They are running in the CI pipeline, see https://github.com/0xE282B0/cobol/runs/7327704765
But the pipeline is not stable enough, yet.

File: exercises/practice/hello-world/src/hello-world.CBL (hunk 3 above)
Comment (KTSnowy) [reply exercises/practice/hello-world/src/hello-world.CBL:8]: That's good to know, I'm glad that it's still working without the `GOBACK`

Could you check something for me, does cobol-check work with free format and lowercase keywords?
You might have to pass the `-free` command line option to gnuCOBOL.
```cobol
*> Sample COBOL program
identification division.
  program-id. hello.

data division.
  working-storage section.
    01 ws-result pic x(13).

procedure division.
  hello-world.
  move "Goodbye, Mars!" to ws-result.
stop run.
            
```

**Review by KTSnowy — APPROVED** [review]
Looks good to me, but we could check if it allows for free format and lowercase keywords for a future PR and to allows us to write COBOL in a more modern way.

**Comment by axtens** [comment]
Regarding free format, gnucobol allows setting it either on the command line or in the source. See https://gnucobol.sourceforge.io/doc/gnucobol.html#Source-format or below
```
2.1.3 Source format

GnuCOBOL supports both fixed and free source format. The default format is the fixed format. This can be overridden either by the >>SOURCE [FORMAT] [IS] {FIXED|FREE} directive, or by one of the following options:

-free, -F

    Free format. The program-text area starts in column 1 and continues till the end of line (effectively 255 characters in GnuCOBOL).
-fixed

    Fixed format. Source code is divided into: columns 1-6, the sequence number area; column 7, the indicator area; columns 8-72, the program-text area; and columns 72-80 as the reference area.[3](https://gnucobol.sourceforge.io/doc/gnucobol.html#FOOT3)
```

**Review by axtens — APPROVED** [review]

File: exercises/practice/hello-world/src/hello-world.CBL (hunk 3 above)
Comment (dakaa16) [reply exercises/practice/hello-world/src/hello-world.CBL:8]: Lower case letters are supported in Cobol check, but free format is not. You can make an issue on that, if it's something you need.
